enigma-swing/src/main/java/cuchaz/enigma/gui/GuiController.java

```diff
@@ -119,6 +133,51 @@ public void closeJar() {
 		this.gui.onCloseJar();
 	}
 
+	public void close() {
+		autoSaveExecutor.shutdown();
+	}
+
+	private synchronized void autoSave() {
+		if (!UiConfig.autoSave() || project == null) {
+			return;
+		}
+
+		final EntryRemapper mapper = project.getMapper();
+		final Path autoSaveDir = ConfigPaths.getConfigPathRoot().resolve("enigma").resolve("autosave");
+
+		try {
+			final Path autoSavePath = autoSaveDir.resolve(DateTimeFormatter.ofPattern("yyyy-MM-dd-HH-mm-ss").format(LocalDateTime.now()) + ".mappings");
+
+			if (Files.notExists(autoSaveDir)) {
+				Files.createDirectories(autoSaveDir);
```

Comment:
I'd just create the directory unconditionally. Doesn't really make a difference though tbh

Reply:
Hmm, what if that name is used by a regular file instead of a directory?

Reply:
I'd just let it fail. Other option would be deleting the file, which is probably a bad idea.

I would probably show a message in the status bar about the autosave status, like "Auto-saved mappings to path /foo/bar" or "Failed to auto-save mappings to /foo/bar", would be a good use of [this](https://github.com/FabricMC/Enigma/blob/master/enigma-swing/src/main/java/cuchaz/enigma/gui/elements/StatusBar.java#L59) instead of just showing it in the log